### PR TITLE
Migrate feature-contextmenu to use browser-state.

### PR DESCRIPTION
**Pull request by pocmo**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [x] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [x] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/ContextMenuFeature.kt [hunk 1]
```diff
@@ -26,66 +36,60 @@ internal const val FRAGMENT_TAG = "mozac_feature_contextmenu_dialog"
  * menu the related [HitResult] will be consumed.
  *
  * @property fragmentManager The [FragmentManager] to be used when displaying a context menu (fragment).
- * @property sessionManager The [SessionManager] instance in order to subscribe to the selected [Session].
+ * @property sessionManager TODO: The [SessionManager] instance in order to subscribe to the selected [Session].
```
Comment (csadilek) [inline components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/ContextMenuFeature.kt]: nit: update kdocs

File: components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/ContextMenuFeature.kt [hunk 2]
```diff
@@ -26,66 +36,60 @@ internal const val FRAGMENT_TAG = "mozac_feature_contextmenu_dialog"
...
  * @property candidates A list of [ContextMenuCandidate] objects. For every observed [HitResult] this feature will query
  * all candidates ([ContextMenuCandidate.showFor]) in order to determine which candidates want to show up in the context
  * menu. If a context menu item was selected by the user the feature will invoke the [ContextMenuCandidate.action]
  * method of the related candidate.
  * @property engineView The [EngineView]] this feature component should show context menus for.
+ * @param customTabId TODO
```
Comment (csadilek) [inline components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/ContextMenuFeature.kt]: nit: update kdocs

File: components/feature/contextmenu/src/test/java/mozilla/components/feature/contextmenu/ContextMenuCandidateTest.kt [hunk 3]
```diff
@@ -237,120 +260,122 @@ class ContextMenuCandidateTest {
...
-        }
-
+        assertNotNull(store.state.tabs.first().content.download)
         assertEquals(
             "https://www.mozilla.org/media/img/logos/firefox/logo-quantum.9c5e96634f92.png",
-            imageDownload!!.url)
+            store.state.tabs.first().content.download!!.url)
     }
 
+    /*
```
Comment (csadilek) [inline components/feature/contextmenu/src/test/java/mozilla/components/feature/contextmenu/ContextMenuCandidateTest.kt]: wow lots of tests to update. What about the ones you commented out? Did you still want to update those or file a follow-up? (happy to help)

Comment (pocmo) [reply components/feature/contextmenu/src/test/java/mozilla/components/feature/contextmenu/ContextMenuCandidateTest.kt]: Oh, oops. I thought I had all of them fixed. Will look at that today.

**Comment by pocmo** [comment]
Updated. And tests complete now.. :)

**Comment by pocmo** [comment]
bors r=csadilek

**Pull request merged**